h5py/tests/test_ros3.py

```diff
@@ -24,22 +24,69 @@
 
 
 @pytest.mark.network
-def test_ros3():
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        pytest.param(
+            {},
+            id="v1",
+            marks=pytest.mark.skipif(
+                h5py.version.hdf5_version_tuple >= (2, 0, 0),
+                reason="Requires HDF5 < 2.0",
+            ),
+        ),
+        pytest.param(
+            {"aws_region": b"us-east-2"},
+            id="v2",
+            marks=pytest.mark.skipif(
+                h5py.version.hdf5_version_tuple < (2, 0, 0),
+                reason="Requires HDF5 >= 2.0",
+            ),
+        ),
+    ],
+)
+def test_ros3(kwargs):
     """ ROS3 driver and options """
 
     with h5py.File("https://dandiarchive.s3.amazonaws.com/ros3test.hdf5", 'r',
-                   driver='ros3') as f:
+                   driver='ros3', **kwargs) as f:
         assert f
         assert 'mydataset' in f.keys()
         assert f["mydataset"].shape == (100,)
 
 
-def test_ros3_s3_fails():
+@pytest.mark.parametrize(
+    "exc,match_exc",
+    [
+        pytest.param(
+            ValueError,
+            [
+                "AWS region required for s3:// location",
+                r"^foo://wrong/scheme: S3 location must begin with",
+            ],
+            id="v1",
```

Comment:
in retrospect, I should have suggested this earlier, sorry for the back-and-forth
```suggestion
            id="HDF5-v1",
```